src/api.py

```diff
@@ -4,31 +4,63 @@
+
+class DeviceResource(Resource):
+    """
+        The device resource handles API requests relating to robomussel data.
+	Will output all device information.
+    """
+    def get(self, site_name):
+   	"""
+            Returns all information about each device 
+   	"""
+	return map(Device.to_json, Device.query.all()) 
 
 class ReadingResource(Resource):
     """
-        The device resources handles API requests relating to robomussel data
+        The reading resources handles API requests relating to robomussel data
         it additionally supports facilities to filter datasets based on query
         string parameters
     """
-    def get(self, dev_name):
+    def get(self):
         """
             Filters the database by the given device id, and returns a JSON
             string to the requester
         """
         return map(Reading.to_json, 
-                self.filter(Reading.query.filter(Reading.device == dev_name)).all())
+                self.filter(Reading.query))
 
     def filter(self, readings):
         """
             Filters the given readings by the parameters encoded in the query
-            string
+            ()string
         """
         args = self.query_parse()
+        
         if args["start_date"] and args["end_date"]:
             readings = readings.filter(Reading.date.between(args["start_date"], 
                                                             args["end_date"]))
+	#http://159.203.111.95:port/api/reading?country="country"
+        if args["country"]:
+	    print("country is %s" % args["country"])
+	    readings = readings.join(Device).filter(Device.country == args["country"])
+        #http://159.203.111.95:port/api/reading?state_province="state_province/country="country"
+        if args["state_province"] and args["country"]:
+            readings = readings.join(Device).filter(Device.state_province == args["state_province"]).filter(Device.country == args["country"])
+        #http://159.203.111.95:port/api/reading?location="location"
+        if args["location"]:
+            readings = readings.join(Device).filter(Device.location == args["location"])        
+        #http://159.203.111.95:port/api/reading?wave_exp="wave_exp"
+        #for layering queries (country and wave_exp) does this make the most sense?
+        if args["country"] and args["wave_exp"]:
+           readings = readings.join(Device).filter(Device.country == args["country"]).filter(Device.wave_exp == args["wave_exp"])
+        #http://159.203.111.95:port/api/reading?device="device"/zone="zone"
+        if args["device"] and args["zone"]:
+           readings = readings.join(Device).filter(Reading.device == args["device"]).filter(Device.zone == args["zone"]) 
+        #http://159.203.111.95:port/api/reading?device="device"/sub_zone="sub_zone"
+        if args["device"] and args["sub_zone"]
+           readings = readings.join(Device).filter(Reading.device == args["device"].filter(Device.sub_zone == args["sub_zone"])
```

Comment:
Love the url examples for every line, super helpful. Might be more readable with a newline before every comment, and a space after the octothorpe. 

Your call.